packages/govuk-frontend/src/govuk/components/header/header.mjs

```diff
@@ -83,23 +83,45 @@ export class Header extends GOVUKFrontendComponent {
     this.$menu = $menu
     this.$menuButton = $menuButton
 
-    // Set the matchMedia to the govuk-frontend desktop breakpoint
-    this.mql = window.matchMedia('(min-width: 48.0625em)')
+    this.setupResponsiveChecks()
+
+    this.$menuButton.addEventListener('click', () =>
+      this.handleMenuButtonClick()
+    )
+  }
+
+  /**
+   * Setup viewport resize check
+   *
+   * @private
+   */
+  setupResponsiveChecks() {
+    const breakpointProperty = '--govuk-frontend-breakpoint-desktop'
+    const breakpointValue = getComputedStyle(document.body).getPropertyValue(
```

Comment:
**nitpick**: `:root` corresponds to `document.documentElement`. It would make the mapping between CSS and JS easier if they both reference the same element, even though the custom property value will cascade down to `document.body`.

Reply:
Ah the issue said `window.getComputedStyle()` and I grabbed [`document.body` from Slack](https://gds.slack.com/archives/C0447D0STV0/p1688038334540579?thread_ts=1688038122.738779&cid=C0447D0STV0)

Shall I use `document.documentElement` for `<html>` instead?

You've suggested `<body>` below in https://github.com/alphagov/govuk-frontend/pull/4562#discussion_r1427030841

Reply:
I've gone with `document.documentElement` to represent `:root`

Reply:
Oh damn sorry for the confusion in the following comment, I did indeed mean to align CSS and JS on the `:root`/`document.documentElement`.